### ISSUE TITLE
1.8.7 compat

### ISSUE THREAD

**Issue by rdp** [arvicco/win#1]
twould be nice :) (I'm stuck with 1.8.7 because jruby doesn't work too well in 1.9 mode [ffi, win32/registry all fail] and that's what I'm using...)
Thanks anyway though, until then I'll just use bits and pieces of your source :)

**Comment by arvicco** [comment]
Yeah, I'm also bumping into all kind of problems with jruby --1.9... Unfortunately, the way this library is structured it won't be easy to backport all the dynamic 1.9 stuff to 1.8.7 - you'll need to rewrite like 80% of it to use 1.8.x-style code evals and other ugly hacks - personally, I'd better wait for jruby 1.9 support to mature. ;)

**Comment by rdp** [comment]
Yeah that would be ideal.  My current hurdle is http://jira.codehaus.org/browse/JRUBY-5216 :)
